packages/nx/src/plugins/js/project-graph/build-dependencies/target-project-locator.ts

```diff
@@ -302,6 +316,52 @@ export class TargetProjectLocator {
     const project = findProjectForPath(file, this.projectRootMappings);
     return this.nodes[project];
   }
+
+  /**
+   * In many cases the package.json will be directly resolvable, so we try that first.
+   * If, however, package exports are used and the package.json is not defined, we will
+   * need to resolve the main entry point of the package and traverse upwards to find the
+   * package.json.
+   *
+   * In some cases, such as when multiple module formats are published, the resolved package.json
+   * might only contain the "type" field - no "name" or "version", so in such cases we keep traversing
+   * until we find a package.json that contains the "name" and "version" fields.
+   */
+  private findExternalPackageJson(
```

Comment:
Should we call this.. `readPackageJson`? 

1. It doesn't just find the file, it also reads it.
2. nit: Do we really need to distinguish it's an external package.json? :shrug:

Reply:
Sure, I just quickly carried it over